### DISCUSSION TITLE
Regarding PMD rules

### DISCUSSION THREAD

**Discussion by jamshiyab** [pmd/pmd#3338] [Q&A] [answered]
Hi,
I am using PMD in command line interface for code analysis of java. Where we can find the dafault rules and in which directory we will create new rules?

**Comment by jsotuyod** [comment, marked as answer]
There are no default rules when running PMD from CLI, the `-rulesets` / `-R` argument is required.

You either [create your own rulest](https://pmd.github.io/latest/pmd_userdocs_making_rulesets.html) or specify a built-in one (such as [`rulesets/java/quickstart.xml`](https://github.com/pmd/pmd/blob/master/pmd-java/src/main/resources/rulesets/java/quickstart.xml)).

As for creating new rules… if you are creating XPath rules, you can define them directly in the ruleset being used. Directory doesn't matter, as you are passing the ruleset file location to the CLI. If you want to write rules in Java, this is slightly different, as you would need to make sure the jar containing the rule is in PMD's classpath when executing it AND make sure to reference such rule from the ruleset XML passed to the CLI.
